### PR TITLE
Implrement Devise

### PR DESCRIPTION
**Pull request by rohalla2**

### CODE REVIEW

**Comment by rohalla2** [comment]
@Stevo393 I'm pretty sure I fixed Master, but this branch is messed up.  Looks like you incorrectly fixed to rebase/merge conflict a couple of times, and I had to mess it up a little more to get things "working."  So, you're going to need to pull this branch down locally and add a new commit that fixes all of the merge problems.  I'll make some comments in the code to show you what the problem actually is.

**Comment by rohalla2** [comment]
Also, until you learn what you're doing with git and github pull requests, don't merge anything into master.  This was a shit storm to undo.  I highly suggest you take a look at some git command line tutorials.  I'm happy to guide you in the meantime.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,11 +1,31 @@
 class UsersController < ApplicationController
+<<<<<<< HEAD
```
Comment (rohalla2) [inline app/controllers/users_controller.rb]: This is a part of a merge conflict that should be removed before completing a rebase.  All of the instances of the following should be removed from the app:
`<<<<<<< HEAD`
Everything between these two tags are what the code is in master.
`=======`
Anything here is the code that you've changed.
`>>>>>>> <Plus some commit hash/message>`

Anywhere where these appear you will need to carefully decide what actually needs to be in the file.  Things got really fucked up, so I can't tell what it supposed to actually be in each file.

Just fix all of these when you can and add them in a new commit.

Comment (Stevo393) [inline app/controllers/users_controller.rb]: Yea, before pushing, i did a search on "<<<<" and went through and did that before pushing. I thought i had committed those changes - i'll have to go back and look.

**Comment by Stevo393** [comment]
I cleaned up the merge conflicts yesterday - just had not commited/pushed. I just pushed the commits managing all the merges.

I went through each of those sections and deleted the unnecessary section and tags --> and still got merge conflicts at the end (as is the case currently on this branch). I get the point of it, it just isn't clear to me how i "resolve" the merge conflicts to satisfy git to do the actual merge.

I'll read the tutorial again for merging/rebasing, i get it intuitively, more of an execution problem at this point.

**Comment by rohalla2** [comment]
@Stevo393 The only problem is that you force pushed from your local branch instead of pulling the fixed branch that I put up here yesterday.  I'll fix this new branch but be aware __You will need to pull this branch before you make any changes__. For the duration of this pull request, if you have to force push, something is probably wrong.

**Comment by rohalla2** [comment]
Closing this PR and opening a new one with the fixed branch.

**Comment by Stevo393** [comment]
I didn't know you working on this branch - so i just copied the previous
command without think. Assuming I had noticed, so would the following be
the best way to handle:

I have commit called "B" (off of my last push "A"), and you have pushed a
commit called "C"

1. Fetch "C"
2. Rebase "B" off of "C" (creating "D")
   2A. This would include Merging the chages between B and C
3. Push "D" which will ofcourse have no conflicts with "C" as its was
rebased off "C"


Which ends up looking like:

A - - - - - C- - - - D
|- - B


Am I thinking of this correctly?

Assuming that is the case, then i need to really understand the best way to
manage (2A). When i previously went through and managed all the merge
conflicts, the issue i had was that commit wasn't a "resolve merge"
conflict. I'm not sure i'm articulating that correctly, so let me know

**Pull request closed without merging**